api/src/AppBundle/Entity/Repository/DocumentRepository.php

```diff
@@ -43,36 +64,46 @@ public function getQueuedDocumentsAndSetToInProgress(string $limit)
 ndr_id,
 ndr_start_date,
 ndr_submit_date,
-report_submission_id,
-report_submission_uuid
+coalesce(report_submission_uuid, all_uuid) as report_submission_uuid
 FROM (
 SELECT DENSE_RANK() OVER(ORDER BY d.is_report_pdf DESC, d.id) AS dn,
 coalesce(c1.case_number, c2.case_number) AS case_number,
-coalesce(rs1.id, rs2.id) AS report_submission_id,
-coalesce(rs1.opg_uuid, rs2.opg_uuid) AS report_submission_uuid,
-d.id AS document_id, d.is_report_pdf, d.filename, d.storage_reference, d.report_submission_id AS document_report_submission_id,
-r.start_date AS report_start_date, r.end_date AS report_end_date, r.submit_date AS report_submit_date, r.type AS report_type,
-o.id AS ndr_id, o.start_date AS ndr_start_date, o.submit_date AS ndr_submit_date
+rs2.opg_uuid as all_uuid,
+rs.opg_uuid AS report_submission_uuid,
+d.id AS document_id,
+d.is_report_pdf,
+d.filename,
+d.storage_reference,
+d.report_submission_id AS document_report_submission_id,
+rs2.id as all_report_submission_id,
+r.start_date AS report_start_date,
+r.end_date AS report_end_date,
+r.submit_date AS report_submit_date,
+r.type AS report_type,
+o.id AS ndr_id,
+o.start_date AS ndr_start_date,
+o.submit_date AS ndr_submit_date
 FROM document d
+LEFT JOIN report_submission rs ON rs.id = d.report_submission_id
+LEFT JOIN report_submission rs2 on rs2.report_id = d.report_id and d.created_on > rs2.created_on
 LEFT JOIN report r ON r.id = d.report_id
 LEFT JOIN odr o ON o.id = d.ndr_id
-LEFT JOIN report_submission rs1 ON rs1.report_id = d.report_id
-LEFT JOIN report_submission rs2 ON rs2.ndr_id = d.ndr_id
 LEFT JOIN client c1 ON c1.id = r.client_id
 LEFT JOIN client c2 ON c2.id = o.client_id
-WHERE d.synchronisation_status = 'QUEUED') AS sub
-WHERE dn < $limit;";
+WHERE d.synchronisation_status = 'QUEUED'
+) AS sub WHERE dn < $limit) as sub2
+WHERE sub2.rown = 1;";
 
         $conn = $this->getEntityManager()->getConnection();
         $stmt = $conn->prepare($queuedDocumentsQuery);
         $stmt->execute();
 
-        $queuedDocumentData = [];
+        $queuedDocumentDatas = [];
```

Comment:
Data is a plural I think

Reply:
You're right - I think I was trying to show that there would be multiple `QueuedDocumentData` objects in the array as opposed to some data about a queued document. Does that make sense!?

Reply:
Yeah I get what you're going for ;)